### PR TITLE
Feature/40 establish system tests

### PR DESCRIPTION
**Pull request by dinjas**
Performs work towards #40 by establishing the capability to add and run system specs. This PR also includes a very simple, example spec (`SomethingSpec`) that should be removed at some point.

### CODE REVIEW

**Comment by nvick** [comment]
We will want to add `SELENIUM_REMOTE_HOST=localhost` to the CircleCI config. https://github.com/OfficeMomsandDads/scheduler/blob/08e8979774e60d52bebea538e608ed25be912a27/.circleci/config.yml#L7-L15

We will also want to add the Selenium image to the CircleCI config as well. `- image: selenium/standalone-chrome` after the Postgres image and config.
https://github.com/OfficeMomsandDads/scheduler/blob/08e8979774e60d52bebea538e608ed25be912a27/.circleci/config.yml#L16-L20

File: Gemfile [hunk 1]
```diff
@@ -85,6 +85,12 @@ group :development do
   gem 'hint-rubocop_style'
 end
 
+group :test do
+  gem 'capybara', '>= 2.15'
+  gem 'selenium-webdriver'
+  gem 'chromedriver-helper'
```
Comment (nvick) [inline Gemfile]: While writing a blog post about this I realized that we don't need the `chromedriver-helper` gem

File: docker-compose.yml [hunk 2]
```diff
@@ -18,6 +19,12 @@ services:
       - WEBPACKER_DEV_SERVER_HOST=webpacker
       - SEED_ADMIN_EMAIL=${SEED_ADMIN_EMAIL}
       - SEED_ADMIN_PASSWORD=${SEED_ADMIN_PASSWORD}
+      - SELENIUM_REMOTE_HOST=selenium
+  selenium:
+    image: selenium/standalone-chrome
+    ports:
+      - 4444:4444
+      - 5900:5900
```
Comment (nvick) [inline docker-compose.yml]: I believe this port is only for the debugging image which we are not using.

File: Gemfile (hunk 1 above)
Comment (nvick) [reply Gemfile]: Since we are using the `selenium/standalone-chrome` container.

**Comment by nvick** [comment]
@benjaminwood when you have a chance we would appreciate you taking a look at this PR.

File: docker-compose.yml [hunk 3]
```diff
@@ -18,6 +19,9 @@ services:
       - WEBPACKER_DEV_SERVER_HOST=webpacker
       - SEED_ADMIN_EMAIL=${SEED_ADMIN_EMAIL}
       - SEED_ADMIN_PASSWORD=${SEED_ADMIN_PASSWORD}
+      - SELENIUM_REMOTE_HOST=selenium
+  selenium:
+    image: selenium/standalone-chrome
```
Comment (benjaminwood) [inline docker-compose.yml]: Slick!

File: spec/support/capybara.rb [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+                             `/sbin/ip route|awk '/scope/ { print $9 }'`.strip
+                           else
+                             '0.0.0.0'
+                           end
+    Capybara.server_port = '43447'
+    Capybara.app_host    = "http://#{Capybara.current_session.server.host}:#{Capybara.current_session.server.port}"
+  end
+
+  config.after :each, type: :system, js: true do
+    page.driver.browser.manage.logs.get(:browser).each do |log|
```
Comment (benjaminwood) [inline spec/support/capybara.rb]: This is interesting, what brought this about?

Comment (nvick) [reply spec/support/capybara.rb]: It was in Noel Rappin’s blog post for using system tests and it seems sane to us so we used it.

**Pull request merged**